nh_clinical/activity_extension.py

```diff
@@ -538,110 +530,241 @@ def get_activity_user_ids(self, cr, uid, activity_id, context=None):
         user_ids += follower_ids
         return list(set(user_ids))
 
-    # TODO EOBS-703: Trigger policy method is too large
-    def trigger_policy(self, cr, uid, activity_id, location_id=None,
-                       case=False, context=None):
+    # TODO EOBS-2171 This can be removed?
+    @api.model
+    def check_policy_activity_context(self, activity, location_id=None):
+        """
+        Check that the policy activity can be triggered based on the location's
+        context (Normally eobs)
+
+        :param activity: Policy activity definition
+        :type activity: dict
+        :param location_id: ID of location to check policy for
+        :type location_id: int
+        :return: True if can create activity for location
+        :rtype: bool
+        """
+        location_model = self.env['nh.clinical.location']
+        triggered_context = activity.get('context')
+        if triggered_context and location_id:
+            location = location_model.browse(location_id)
+            # Create a list of bools representing if the activity and location
+            # contexts match. If none of them match then return False
+            if not any(
+                    [c.name == triggered_context for c in
+                     location.context_ids]):
+                return False
+        return True
+
+    @api.model
+    def check_trigger_domains(self, activity, domains=None):
+        """
+        Check if any records are returned for the supplied domains. If so then
+        return False
+
+        :param activity: nh.clinical.activity instance
+        :param domains: list of dicts with object (model to query) and domain
+            (search query to carry out on model)
+        :return: True if no records found, False is records found
+        :rtype: bool
+        """
+        if not domains:
+            return False
+        for domain in domains:
+            domain_pool = self.env[domain['object']]
+            search_domain = domain['domain'] + [
+                ['parent_id', '=', activity.spell_activity_id.id]
+            ]
+            if domain_pool.search(search_domain):
+                return True
+        return False
```

Comment:
Again I'm confused by variable names. I thought a _domain_ was a parameter in a search for Odoo records but here its referring to a dictionary that contains a model name and a search parameter that I would usually call a _domain_. There's also something called a `domain_pool` which makes no sense to me.